lib/Driver/UnixToolChains.cpp

```diff
@@ -165,7 +168,10 @@ toolchains::GenericUnix::constructInvocation(const DynamicLinkJobAction &job,
 
   // Select the linker to use.
   std::string Linker;
-  if (const Arg *A = context.Args.getLastArg(options::OPT_use_ld)) {
+  if (context.OI.LTOVariant != OutputInfo::LTOKind::None) {
+    // Force to use lld for LTO
+    Linker = "lld";
+  } else if (const Arg *A = context.Args.getLastArg(options::OPT_use_ld)) {
```

Comment:
Either `OPT_use_ld` should take precedence over this LTO-specific branch, or the combination of `OPT_use_ld` and `OPT_lto` should diagnose an error. Silently ignoring `-use-ld` is a bad move.

Reply:
OK, I changed to prefer `-use-ld` option.